scalafmt-sbt/src/main/scala/org/scalafmt/sbt/ScalafmtPlugin.scala

```diff
@@ -1,25 +1,169 @@
+  private type Output = String
+
+  private def withFormattedSources[T](
+      sources: Seq[File],
+      config: ScalafmtConfig
+  )(
+      onError: (File, Throwable) => T,
+      onFormat: (File, Input, Output) => T
+  ): Seq[Option[T]] = {
+    sources.map(
+      file => {
+        val input = IO.read(file)
+        val output = Scalafmt.format(input, config)
+
+        output match {
+          case Formatted.Failure(e) =>
+            if (config.runner.fatalWarnings) {
+              throw e
+            } else if (config.runner.ignoreWarnings) {
+              // do nothing
+              None
+            } else {
+              Some(onError(file, e))
+            }
+          case Formatted.Success(code) =>
+            Some(onFormat(file, input, code))
+        }
       }
+    )
   }
-  override def globalSettings: Seq[Def.Setting[_]] =
-    Seq(
-      commands += autoImport.scalafmt
-    ) ++
-      addCommandAlias("scalafmtTest", "scalafmt --test") ++
-      addCommandAlias("scalafmtDiffTest", "scalafmt --diff --test") ++
-      addCommandAlias("scalafmtDiff", "scalafmt --diff")
 
+  private def formatSources(
+      sources: Seq[File],
+      config: ScalafmtConfig,
+      log: Logger
+  ): Unit = {
+    withFormattedSources(sources, config)(
+      (file, e) => log.error(s"Error in ${file.toString}: $e"),
+      (file, input, output) => {
+        if (input != output) {
+          log.info(s"Successfully formatted ${file.toString}.")
+          IO.write(file, output)
+        }
+      }
+    )
+  }
+
+  private def checkSources(
+      sources: Seq[File],
+      config: ScalafmtConfig,
+      log: Logger
+  ): Boolean = {
+    withFormattedSources(sources, config)(
+      (file, e) => {
+        log.error(s"Error in ${file.toString}: $e")
+        false
+      },
+      (file, input, output) => {
+        val diff = input != output
+        if (diff) {
+          log.error(s"${file.toString} isn't formatted properly!")
+        }
+        diff
+      }
+    ).flatten.forall(x => x)
+  }
+
+  private lazy val sbtSources = thisProject.map(
+    proj => {
+      val rootSbt =
+        BuildPaths.configurationSources(proj.base).filterNot(_.isHidden)
+      val projectSbt =
+        (BuildPaths.projectStandard(proj.base) * GlobFilter("*.sbt")).get
+          .filterNot(_.isHidden)
+      rootSbt ++ projectSbt
+    }
+  )
+  private lazy val projectSources = thisProject.map(proj =>
+    (BuildPaths.projectStandard(proj.base) * GlobFilter("*.scala")).get)
+
+  lazy val scalafmtSettings: Seq[Def.Setting[_]] = Seq(
+    scalafmtOnCompile := false,
+    scalafmt := formatSources(
+      unmanagedSources.value,
+      scalaConfig.value,
+      streams.value.log),
+    scalafmtSbt := {
+      formatSources(sbtSources.value, sbtConfig.value, streams.value.log)
+      formatSources(projectSources.value, scalaConfig.value, streams.value.log)
+    },
+    scalafmtCheck :=
+      checkSources(
+        unmanagedSources.value,
```

Comment:
Same: `(unmanagedSources in scalafmt).value` (not `in scalafmtCheck` in this case, because `scalafmtCheck` should always apply on the same set of sources as `scalafmt`).